repos/base-linux/src/core/platform_services.cc

```diff
@@ -0,0 +1,38 @@
+/*
+ * \brief  Platform specific services for x86
+ * \author Stefan Kalkowski
+ * \date   2012-10-26
+ */
+
+/*
+ * Copyright (C) 2012-2017 Genode Labs GmbH
+ *
+ * This file is part of the Genode OS framework, which is distributed
+ * under the terms of the GNU Affero General Public License version 3.
+ */
+
+/* Genode includes */
+#include <base/service.h>
+
+/* core includes */
+#include <core_env.h>
+#include <platform.h>
+#include <platform_services.h>
+#include <io_port_root.h>
+
+#include <core_linux_syscalls.h>
+
+/*
+ * Add x86 specific ioport service
+ */
+void Genode::platform_add_local_services(Rpc_entrypoint*,
+                                         Sliced_heap *sliced_heap,
+                                         Registry<Service> *local_services)
+{
+        lx_iopl(3);
```

Comment:
As discussed f2f today: check iopl() return value and execute below code only on success. The session should be denied in this case.